README.md

```
@@ -0,0 +1,216 @@
+
+Basic rules provide rule criteria and a priority value to use when applying the rule.
+
+Advanced rules can either provide a priority value or specify "avoid" action to prevent the router form picking a road.
+
+Priority values must be between 0 (meaning no priority change) and 255 (highest priority). When a possible road is evaluated, all priority values are summed up from all rules and the one with the highest total priority is picked. If a single "avoid" action is encountered, the road is excluded regardless of the priority values.
+
+An example rule file can be found in `./rule-examples/rules-prefer-unpaved.json` that prefers smaller unpaved roads.
+
+An example rule file that will not pick unpaved roads or paths and trails can be seen here `./rule-examples/rules-avoid-unpaved.json`
+
+Road types, smoothness and surfaces are based on OpenStreetMap.org tag values. Road type is specified as "highway" (https://wiki.openstreetmap.org/wiki/Key:highway), while smoothness (https://wiki.openstreetmap.org/wiki/Key:smoothness) and surface (https://wiki.openstreetmap.org/wiki/Key:surface) are specified as such.
```

Markdownlint (0.37.0)

Bare URL used

(MD034, no-bare-urls)